static/app/components/profiling/boundTooltip.tsx

```diff
@@ -6,39 +6,76 @@ import {space} from 'sentry/styles/space';
 import type {CanvasView} from 'sentry/utils/profiling/canvasView';
 import {useFlamegraphTheme} from 'sentry/utils/profiling/flamegraph/useFlamegraphTheme';
 import type {FlamegraphCanvas} from 'sentry/utils/profiling/flamegraphCanvas';
-import type {Rect} from 'sentry/utils/profiling/speedscope';
+import {Rect} from 'sentry/utils/profiling/speedscope';
 import theme from 'sentry/utils/theme';
 
+// The cursor icon is drawn with an origin in the top left, which means that if we render
+// a tooltip directly at the cursor's position, it will overlap with the cursor icon.
+//          x      <- client x
+//          |----| <- cursor icon
+//          |-------------| <- tooltip
+//
+// This wont happen if we draw the tooltip to the left of the cursor, as the cursor icon will
+// be drawn to right of the tooltip. The offset helps us correct this and remove the overlap.
+//               x      <- client x offset
+//               |----| <- cursor icon
+// |-------------|      <- tooltip
+
+// We only need to do this when drawing the tooltip on the left side of the cursor, as
+// the origin is in the correct position when drawing the tooltip on the right side.
+const CURSOR_LEFT_OFFSET_PX = 6;
+const CURSOR_TOP_OFFSET_PX = 4;
+// Gap between the tooltip and container edge for each side
+const WIDTH_OFFSET = 8;
+
 function computeBestTooltipPlacement(
   cursor: vec2,
-  container: Rect,
-  tooltip: DOMRect
+  tooltip: DOMRect,
+  canvas: Rect,
+  container: Rect
 ): string {
   // This is because the cursor's origin is in the top left corner of the arrow, so we want
   // to offset it just enough so that the tooltip does not overlap with the arrow's tail.
   // When the tooltip placed to the left of the cursor, we do not have that issue and hence
   // no offset is applied.
-  const OFFSET_PX = 6;
-  let left = cursor[0] + OFFSET_PX;
-  const top = cursor[1] + OFFSET_PX;
+  const cursorLeft = cursor[0];
+  const cursorTop = cursor[1];
+
+  // Cursor is relative to canvas, not container
+  const cursorRelativeToContainer = cursorLeft + canvas.x;
+
+  let left = cursorLeft;
+
+  left =
+    cursorRelativeToContainer > container.width / 2
+      ? cursorLeft - tooltip.width
+      : (left += CURSOR_LEFT_OFFSET_PX);
+
+  const right = left + tooltip.width + canvas.left;
 
-  if (cursor[0] > container.width / 2) {
-    left = cursor[0] - tooltip.width; // No offset is applied here as tooltip is placed to the left
+  if (left + canvas.left - WIDTH_OFFSET <= 0) {
+    left = -canvas.left + WIDTH_OFFSET;
+  } else if (right >= container.width - WIDTH_OFFSET) {
+    left = container.width - tooltip.width - canvas.left - WIDTH_OFFSET;
   }
 
-  return `translate(${left || 0}px, ${top || 0}px)`;
+  return `translate(${left}px, ${cursorTop + CURSOR_TOP_OFFSET_PX}px)`;
```

Comment:
A comment explaining the final `left` is relative to the canvas would be helpful here.

Reply:
Yeah, I added this where we perform the bound checks